src/query/utils.rs

```diff
@@ -41,6 +44,55 @@ where
         .and_then(|transport_cell| from_connection(&transport_cell, compression))
 }
 
+pub fn send_frame_to_all_connections<S, T, M>(sender: &S, frame_bytes: Vec<u8>) -> error::Result<Frame>
+    where
+        S: GetAllConnections<T, M> + GetCompressor<'static> + Sized,
+        T: CDRSTransport + 'static,
+        M: r2d2::ManageConnection<Connection=RefCell<T>, Error=error::Error> + Sized,
+{
+    let ref compression = sender.get_compressor();
+
+    let mut results: Vec<error::Result<Frame>> = Vec::new();
+
+    for connection in sender.get_all_connections() {
+        match connection {
+            Some(conn) => {
+                results.push(write_frame_to_connection(&frame_bytes, compression, conn));
+            }
+            None => { results.push(transform_error()); }
+        }
+    }
+
+    let result = results.iter().find(|r| r.is_err())
+        .unwrap_or_else( ||get_any_valid_result(&results));
+
+    match result {
```

Comment:
Perhaps this statement can be simplified so that no clone is needed. Please consider following line to put instead the whole `match`.

```rust
result.map_err(|err| error::Error::General(e.description().to_string())
```